packages/client/src/views/Settings/PasswordChangeModals.tsx

```diff
@@ -0,0 +1,417 @@
+import { EMPTY_STRING } from '@client/utils/constants'
+import { UserDetails } from '@client/utils/userUtils'
+import { Dialog } from '@opencrvs/components/lib/Dialog/Dialog'
+import * as React from 'react'
+import { injectIntl, WrappedComponentProps as IntlShapeProps } from 'react-intl'
+import { connect } from 'react-redux'
+import { buttonMessages, userMessages as messages } from '@client/i18n/messages'
+import { Button } from '@opencrvs/components/lib/Button'
+import { Text } from '@opencrvs/components/lib/Text'
+import styled from 'styled-components'
+import { InputField } from '@opencrvs/components/lib/InputField'
+import { TextInput } from '@opencrvs/components/lib/TextInput'
+import { TickOff, TickOn } from '@opencrvs/components/lib/icons'
+
+const Message = styled.div`
+  margin-bottom: 16px;
+`
+
+const PasswordContents = styled.div`
+  color: ${({ theme }) => theme.colors.copy};
+  max-width: 100%;
+  & input {
+    width: 100%;
+  }
+  @media (max-width: ${({ theme }) => theme.grid.breakpoints.lg}px) {
+    max-width: 100%;
+  }
+  @media (max-width: ${({ theme }) => theme.grid.breakpoints.md}px) {
+    width: 100%;
+  }
+`
+const ValidationRulesSection = styled.div`
+  background: ${({ theme }) => theme.colors.background};
+  margin: 16px 0 24px;
+  padding: 8px 24px;
+  & div {
+    padding: 8px 0;
+    display: flex;
+    align-items: center;
+    & span {
+      margin-left: 8px;
+    }
+  }
+  @media (min-width: ${({ theme }) => theme.grid.breakpoints.lg}px) {
+    display: none;
+  }
+`
+const ValidationRulesSectionLg = styled.div`
+  background: ${({ theme }) => theme.colors.background};
+  margin: 30px 0px;
+  width: 100%;
+  padding: 8px 24px;
+  & div {
+    padding: 8px 0;
+    display: flex;
+    align-items: center;
+    & span {
+      margin-left: 8px;
+    }
+  }
+  @media (max-width: ${({ theme }) => theme.grid.breakpoints.lg}px) {
+    display: none;
+  }
+`
+const PasswordMatch = styled.div`
+  ${({ theme }) => theme.fonts.bold16};
+  color: ${({ theme }) => theme.colors.positive};
+  margin-top: 8px;
+`
+const PasswordMismatch = styled.div`
+  ${({ theme }) => theme.fonts.bold16};
+  color: ${({ theme }) => theme.colors.negative};
+  margin-top: 8px;
+`
+
+const Field = styled.div`
+  margin-bottom: 30px;
+  @media (max-width: ${({ theme }) => theme.grid.breakpoints.lg}px) {
+    margin-bottom: 0px;
+  }
+`
+
+type State = {
+  currentPassword: string
+  incorrectCurrentPassword: boolean
+  newPassword: string
+  confirmPassword: string
+  validLength: boolean
+  hasNumber: boolean
+  hasCases: boolean
+  passwordMismatched: boolean
+  passwordMatched: boolean
+  errorOccured: boolean
+  currentStepModal: 1 | 2
+}
+
+interface IProps {
+  showPasswordChange: boolean
+  togglePasswordChangeModal: () => void
+  passwordChanged: () => void
+  userDetails: UserDetails | null
+}
+type IFullProps = IProps & IntlShapeProps
+
+class PasswordChangeModalsComp extends React.Component<IFullProps, State> {
```

Comment:
@wilrona could you convert this to a function component? We're slowly trying to get rid of all class components and don't want to introduce a new one